backend/h5p/H5PContentStorage.ts

```diff
@@ -0,0 +1,453 @@
+                LoernwerkErrorCodes.FORBIDDEN
+            );
+        }
+
+        await DBH5PFile.delete({
+            ownerType: 'content',
+            owner: contentId,
+            filename: filename,
+        });
+    }
+
+    /**
+     * Checks if a file exists.
+     * @param contentId The id of the content to add the file to
+     * @param filename the filename of the file to get; can be a path including
+     * subdirectories (e.g. 'images/xyz.png')
+     * @returns true if the file exists
+     */
+    async fileExists(contentId: ContentId, filename: string): Promise<boolean> {
+        const file = await DBH5PFile.findOneBy({
+            ownerType: 'content',
+            owner: contentId,
+            filename: filename,
+        });
+        return file !== null;
+    }
+
+    /**
+     * Returns information about a content file (e.g. image or video) inside a
+     * piece of content.
+     * @param contentId the id of the content object that the file is attached to
+     * @param filename the filename of the file to get information about
+     * @param user the user who wants to retrieve the content file
+     * @returns FileStats of the requested file
+     */
+    async getFileStats(
+        contentId: ContentId,
+        filename: string,
+        user: IUser
+    ): Promise<IFileStats> {
+        if (!(await this.canUserAccessContent(user.id, contentId, true))) {
+            throw new LoernwerkError(
+                'User isnt allowed to edit this content',
+                LoernwerkErrorCodes.FORBIDDEN
+            );
+        }
+
+        return await DBH5PFile.findOne({
+            select: { birthtime: true, size: true },
+            where: {
+                ownerType: 'content',
+                owner: contentId,
+                filename: filename,
+            },
+        });
+    }
+
+    /**
+     * Returns a readable stream of a content file (e.g. image or video) inside
+     * a piece of content
+     * @param contentId the id of the content object that the file is attached to
+     * @param filename the filename of the file to get; can be a path including
+     * subdirectories (e.g. 'images/xyz.png')
+     * @param user the user who wants to retrieve the content file
+     * @param rangeStart (optional) the position in bytes at which the stream
+     * should start
+     * @param rangeEnd (optional) the position in bytes at which the stream
+     * should end
+     * @returns the stream (that can be used to send the file to the user)
+     */
+    async getFileStream(
+        contentId: ContentId,
+        filename: string,
+        user: IUser,
+        rangeStart?: number,
+        rangeEnd?: number
+    ): Promise<Readable> {
+        const file = await DBH5PFile.findOneBy({
+            ownerType: 'content',
+            owner: contentId,
+            filename: filename,
+        });
+        let read: Readable;
+        if (rangeStart !== undefined && rangeEnd !== undefined) {
```

Comment:
I think it could be better if this was done by either trinary operators or `range | 0`